centroid_decompostition.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+#include <list>
+
+using namespace std;
+
+struct Graph 
+{
+	int V;
+	list<int> *adj;
+	int *size;
```

Comment:
Don't use pointers for any of these three variables, especially the list (Wrap everything in a `std::vector`). Also, a list seems overkill for this project. Use `std::vector` for that as well. Also, the variable names should be plural, otherwise this is misleading.